core/templates/domain/classroom/classroom-backend-api.service.ts

```diff
@@ -127,7 +127,7 @@ export class ClassroomBackendApiService {
       }
     }, errorResponse => {
       if (errorCallback) {
-        errorCallback(errorResponse.error.error);
+        errorCallback(errorResponse?.error?.error);
```

Comment:
a> Why the optional chaining is used here?
b> why it is not required earlier, (this is throwing any error)?

same for all changes

Reply:
a> Optional chaining (?.) is like a safety net. It helps access properties of an object without worrying if the object or its properties exist or not. If they don’t exist, it simply gives us undefined instead of causing an error.

b> Earlier, the code worked without optional chaining because the properties we needed were always available. Now, we’ve found cases where errorResponse or errorResponse.error might be null or undefined. i think the only way i thought would be prevent by optional chaining 

Thanks